src/base_client.py

```
@@ -0,0 +1,102 @@
+        
+        except requests.exceptions.HTTPError as e:
+            self.logger.error(f"HTTP error occurred: {e}")
+            raise RuntimeError(f"HTTP error: {e}")
+        
+        except requests.exceptions.RequestException as e:
+            self.logger.error(f"Network error occurred: {e}")
+            raise RuntimeError(f"Network error: {e}")
+        
+        except ValueError as e:
+            self.logger.error(f"JSON parsing error: {e}")
+            raise RuntimeError(f"Could not parse response: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)